BedrockServer.cpp

```diff
@@ -380,10 +418,18 @@ void BedrockServer::sync(const SData& args,
             // (for instance, adding an index).
             upgradeInProgress.store(true);
             if (server._upgradeDB(db)) {
-                server._syncThreadCommitMutex.lock();
+                try {
+                    SDEBUG("Locking the _syncThreadCommitMutex for _upgradeDB");
+                    server._syncThreadCommitMutex.lock();
+                } catch(const std::system_error& e){
```

Comment:
This allows us to capture the system error code, and makes the failure clearer IMO, but I'm happy to remove it since the behavior is the same and we'll still get a stack trace.